packages/auth/amplify_auth_cognito_dart/lib/src/util/aws_iam_auth_provider.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@internal
+library amplify_auth_cognito.util.authorization_providers;
+
+import 'dart:async';
+
+import 'package:amplify_auth_cognito_dart/amplify_auth_cognito_dart.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:aws_signature_v4/aws_signature_v4.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+/// [AmplifyAuthProvider] implementation that signs a request using AWS credentials
+/// from `Amplify.Auth.fetchAuthSession()` or allows getting credentials directly.
+@internal
+class AWSIAMAuthProvider extends AWSCredentialsAmplifyAuthProvider {
```

Comment:
Dart convention is to capitalize acronyms of 2 letters, but pascal case everything else: https://dart.dev/guides/language/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words

My convention is to follow this for everything but `AWS`, which I think looks better capitalized, but only when it's the first word in a name. So this should be `AWSIamAuthProvider`